src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -397,12 +399,52 @@ export class NfsFormComponent extends CdForm implements OnInit {
       }
       client.addresses = addressStr;
     });
-
     this.nfsForm.patchValue(res);
+    if (this.isEdit) {
+      this.resolveSubvolGrp(res);
+    }
     this.setPathValidation();
     this.clients = res.clients;
   }
 
+  resolveSubvolGrp(res: any) {
+    const splitPath = res.path.split('/');
+    const subvolumeGroup = splitPath[2];
+    const subvolume = splitPath[3];
+    const fs_name = this.nfsForm.getValue('fsal').fs_name;
+    this.subvolgrpService.get(fs_name).subscribe((data: CephfsSubvolumeGroup[]) => {
+      this.allsubvolgrps = data;
+      if (subvolumeGroup) {
+        const subvolGrp = this.allsubvolgrps.find((group) => group.name === subvolumeGroup);
+        if (subvolGrp) {
+          this.nfsForm.get('subvolume_group').setValue(subvolGrp.name);
+        }
+        this.resolveSubvol(fs_name, subvolume, subvolumeGroup);
+      }
```

Comment:
Shouldn't this be sufficient, if the data is not editable? Same for subvolume
```suggestion
        if (subvolGrp) {
          this.nfsForm.get('subvolume_group').setValue(subvolGrp.name);
        }```

Reply:
We will need to filter out subvolGrp and resolveSubVol function will also be required.